app/Models/UserAccountHistory.php

```diff
@@ -68,18 +71,19 @@ public static function logUserUpdateEmail($user, $previousEmail)
 
     public function scopeBans($query)
     {
-        return $query->where('ban_status', '<>', static::TYPES['note'])->orderBy('timestamp', 'desc');
+        return $query->whereIn('ban_status', [static::TYPES['restriction'], static::TYPES['silence']])->orderBy('timestamp', 'desc');
     }
 
     public function scopeDefault($query)
     {
-        return $query->where('ban_status', static::TYPES['silence']);
+        return $query->whereIn('ban_status', [static::TYPES['silence'], static::TYPES['tournament_ban']]);
     }
 
     public function scopeRecent($query)
     {
         return $query
             ->where('timestamp', '>', Carbon::now()->subDays(config('osu.user.ban_persist_days')))
+            ->orWhere('permanent', true)
```

Comment:
this may cause rare problem when someone has hundreds of account history entries because their whole history will have to be checked. I'm not quite sure how much impact it'll have but it's probably/hopefully low enough it may be fine.

Otherwise some fancy indexing setup will be required.

Reply:
```sql
mysql> select count(ban_id) from osu_user_banhistory group by user_id order by count(ban_id) desc limit 10;
+---------------+
| count(ban_id) |
+---------------+
|          3324 |
|          1650 |
|          1512 |
|           889 |
|           885 |
|           804 |
|           661 |
|           551 |
|           444 |
|           395 |
+---------------+
```

Possibly a valid concern? Not sure. Could likely be enough to add a new index on `(user_id,permanent)` if we go with that.

Reply:
That doesn't work because the query is an `or` between `permanent` and `timestamp` (and then ordered by `timestamp`).

Reply:
Hmm.. usually mysql will do a collate index usage, but I guess it doesn't bother due to the final `ORDER BY`? (it might also change depending on the number of rows..)

If that's the case it's probably not worth worrying about until it becomes an issue... although it's one of those issue we will likely never notice until it's so bad it takes things down.